Defend The Divine/Assets/Scripts/WaveManager.cs

```diff
@@ -4,60 +4,72 @@
 
 public class WaveManager : MonoBehaviour
 {
-    // TODO: Make array
-    public GameObject enemyPrefab;
+    // Can be made into an array if necessary
+    public GameObject[] enemyPrefabs = new GameObject[3];
 
     [SerializeField]
     private Transform[] enemySpawnPositions;
     public Transform[] EnemySpawnPositions { get { return enemySpawnPositions; } }
-
-
-    /* TEMP VARIABLES FOR TEMP ENEMY SPAWNING */
-    //private float spawnDelay = 1f;
-    //private float spawnTimer = 0;
+    
+    [SerializeField]
+    private int currentWave;
+    private const int MinTimeBetweenWaves = 10;
+    private float waveTimestamp = -(MinTimeBetweenWaves/2);
+    
 
     private void Awake() {
         if (enemySpawnPositions.Length == 0) {
-            Debug.LogError("Add enemy spawn positions to array");
+            Debug.LogError("Enemy spawn positions array is empty in WaveManager");
         }
-        SpawnEnemyGroups(enemyPrefab, 5, 5000, 10);
-        SpawnEnemyGroups(enemyPrefab, 5, 3000, 10);
+        currentWave = 0;
     }
 
     void Update()
     {
-/*        // Temp spawn enemy every 5 seconds
-        spawnTimer += Time.deltaTime;
-        if (spawnTimer > spawnDelay)
+        if (GameManager.Instance.enemies.Count < 1 &&
+            Time.realtimeSinceStartup > MinTimeBetweenWaves + waveTimestamp)
```

Comment:
I'm curious, isn't enemies.Count always greater than 1? Because even if we remove an enemy from the enemyList, isn't it still null there?

Reply:
Nope, enemies.Count represents the number of living enemies, as when an enemy dies, it calls 
`GameManager.Instance.RemoveEnemy(this);`
which is a GameManager function that calls
`enemies.[Remove](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.remove?view=net-7.0)(this)`